cli/pcluster/utils.py

```diff
@@ -290,6 +290,87 @@ def get_supported_compute_instance_types(scheduler):
     return instances
 
 
+def get_supported_az_for_instance_type(instance_type):
+    """
+    Return a tuple of availability zones that have the instance_types.
+
+    This function build above get_supported_az_for_instance_types,
+    but simplify the input to 1 instance type and result to a list
+
+    :param instance_type: the instance type for which the supporting AZs.
+    :return: a tuple of the supporting AZs
+    """
+    return get_supported_az_for_instance_types([instance_type])[instance_type]
+
+
+def get_supported_az_for_instance_types(instance_types):
```

Comment:
Again, caching looks a bit complex here without adding any real benefit, let's remove it.

I don't think we need `get_supported_az_for_instance_type ` as well? Maybe just keep `get_supported_az_for_instance_types ` function as the only function to call boto3 and get list of qualified azs for instances. We can store the result in variables, say `qualified_master_azs` and `qualified_compute_azs`, and just pass them around

Reply:
Right. Although the current approach add some complexity in these utility functions, it makes code invoking them less complex. `get_suppported_az_for_instance_type` is an attempt to make code elsewhere shorter